WalletWasabi.Fluent/ViewModels/Dialogs/InsufficientBalanceDialogViewModel.cs

```diff
@@ -1,28 +1,188 @@
+	{
+		_transactionInfo.SubtractFee = true;
+		Close();
+	}
+
+	private void OnSendAnyway()
+	{
+		_transactionInfo.MaximumPossibleFeeRate = _maximumPossibleFeeRate;
+		_transactionInfo.FeeRate = _maximumPossibleFeeRate;
+		_transactionInfo.ConfirmationTimeSpan = _confirmationTimeWithMaxFeeRate;
+		Close();
+	}
+
+	protected override void OnNavigatedTo(bool isInHistory, CompositeDisposable disposables)
+	{
+		base.OnNavigatedTo(isInHistory, disposables);
+
+		if (!isInHistory)
+		{
+			RxApp.MainThreadScheduler.Schedule(async () =>
+			{
+				// Edge cases, probably will never happen.
+				if (!EnableSendAnyway && !EnableSubtractFee && !EnableSelectMoreCoins)
+				{
+					await ShowErrorAsync("Transaction Building",
+						"Automatic transaction fee selection is not possible at the moment. Alternatively, you can enter the transaction fee manually in Advanced options.",
+						"Wasabi was unable to create your transaction.");
+					Close(DialogResultKind.Back);
+					return;
+				}
+
+				if (_transactionInfo.IsPayJoin && _wallet.Coins.TotalAmount() == _transactionInfo.Amount)
+				{
+					await ShowErrorAsync("Transaction Building", "There are not enough funds to cover the transaction fee.", "Wasabi was unable to create your transaction.");
+					Close(DialogResultKind.Back);
+					return;
+				}
+
+				IsBusy = true;
+				await CheckSilentCaseAsync(_differenceOfFeePercentage, _wallet, _transactionInfo);
+				SetQuestion();
+				IsBusy = false;
+			});
+		}
+	}
+
+	private async Task CheckSilentCaseAsync(decimal percentage, Wallet wallet, TransactionInfo transactionInfo)
+	{
+		if (percentage == 0)
+		{
+			return;
+		}
+
+		if (!TransactionFeeHelper.TryGetMaximumPossibleFeeRate(percentage, wallet, transactionInfo.FeeRate, out var maxFeeRate))
+		{
+			await ShowErrorAsync("Transaction Building",
+				"Automatic transaction fee selection is not possible at the moment. Alternatively, you can enter the transaction fee manually in Advanced options.",
+				"Wasabi was unable to create your transaction.");
+			Close(DialogResultKind.Back);
+			return;
+		}
+
+		_maximumPossibleFeeRate = maxFeeRate;
+		_confirmationTimeWithMaxFeeRate = TransactionFeeHelper.CalculateConfirmationTime(maxFeeRate, wallet);
+
+		if (percentage <= TransactionFeeHelper.FeePercentageThreshold)
+		{
+			OnSendAnyway();
+		}
 	}
 
-	public string Question { get; }
+	private void SetQuestion()
+	{
+		var sendingTotalBalance = _wallet.Coins.TotalAmount() == _transactionInfo.Amount;
+
+		Question = $"There are not enough funds {(!sendingTotalBalance ? "selected " : "")}to cover the preferred transaction fee. Alternatively, ";
+
+		if (EnableSubtractFee)
+		{
+			Question += "you can subtract the transaction fee from the amount.";
+			return;
+		}
+
+		if (EnableSendAnyway)
+		{
+			Question += $"you can send the transaction with an adjusted transaction fee, so it should be confirmed within approximately {TextHelpers.TimeSpanToFriendlyString(_confirmationTimeWithMaxFeeRate)}";
```

Comment:
I think this clarifies better what you mean by `adjusted transaction fee`

```suggestion
			Question += $"you can send the transaction with a maximum possible transaction fee, so it should be confirmed within approximately {TextHelpers.TimeSpanToFriendlyString(_confirmationTimeWithMaxFeeRate)}";
```

Reply:
@danwalmsley will tweak the wording soon, wait for it first.